### ISSUE TITLE
AWS::Serverless::Api Auth, ApiKeyRequired, and UsagePlan property resource generation is broken for me

### ISSUE THREAD

**Issue by shivatalwar** [aws/serverless-application-model#1665]
**Description:**
```
    ServerlessRestApi:
      Type: AWS::Serverless::Api
      Properties:
        StageName: Prod
        Auth:
          ApiKeyRequired: true
          UsagePlan:
            CreateUsagePlan: PER_API
            Quota:
              Limit: 33333
              Period: DAY
```
**Observed result:**
This does not create any UsagePlan, ApiKey, or UsagePlanKey resources. This also does not set APIKey required on any methods on a fresh deploy. Is this intentional? I have tried using both sam and cloudformation package and deploy CLI tooling. 
The deployed template shows this exact representation on the console, with none of these in the resources tab or in the API Gateway console.

**Expected result:**
I expect the UsagePlan, ApiKey, or UsagePlanKey resources to be provisioned, and each method in my API in this stage to require an APIKey. 

**Separately/ As an aside**, this type of resource generation bug has also occurred surrounding API DomainName configuration. Specifically, I've had to link BasePathMappings and DomainNames using native CloudFormation because of these regressions in Serverless Transforms.

**Comment by omardoma** [comment]
Same issue for me, all of these AWS::Serverless::Api are not working/being created:

- Auth.ApiKeyRequired
- DisableExecuteApiEndpoint
- Models
- Domain

**Comment by GavinZZ** [comment]
Hi there was [another issue](https://github.com/aws/serverless-application-model/issues/984) which sounds very similar to this problem where when using `SAM Function` event source with `Authorizers, ApiKeyRequired` properties along with an explicitly defined `SAM API`, we noticed that it doesn't properly apply the `Authorizers, ApiKeyRequired` fields. 

We recently fixed this problem by introducing this new property [OverrideApiAuth](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-function-apifunctionauth.html#sam-function-apifunctionauth-overrideapiauth). When setting it to true, it would fix the issues I described above. A more detailed explanation is provided in the documentation.

Feel free to give it a try and re-open this ticket if it doesn't fix your problem.
